pkg/miniogw/integration_test.go

```diff
@@ -0,0 +1,166 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package miniogw
+
+import (
+	"context"
+	"flag"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/minio/cli"
+	minio "github.com/minio/minio/cmd"
+	"github.com/stretchr/testify/assert"
+
+	"storj.io/storj/internal/identity"
+	"storj.io/storj/internal/s3client"
+	"storj.io/storj/internal/testcontext"
+	"storj.io/storj/internal/testplanet"
+	"storj.io/storj/pkg/cfgstruct"
+	"storj.io/storj/pkg/miniogw/logging"
+	"storj.io/storj/pkg/provider"
+)
+
+func TestUploadDownload(t *testing.T) {
+	var (
+		apiKey     = "apiKey"
+		encKey     = "encKey"
+		bucket     = "bucket"
+		objectName = "testdata"
+		ctx = testcontext.New(t)
+	)
+
+	planet, err := testplanet.New(t, 1, 30, 0)
+	assert.NoError(t, err)
+
+	defer func() {
+		err = planet.Shutdown()
+		if err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	// create temporary directory for minio
+	tmpDir := ctx.Dir()
+	
+	defer ctx.Cleanup()
+
+	err = flag.Set("pointer-db.auth.api-key", apiKey)
+	assert.NoError(t, err)
+
+	// bind default values to config
+	var gwCfg Config
+	cfgstruct.Bind(&flag.FlagSet{}, &gwCfg)
+
+	// minio config directory
+	gwCfg.MinioDir = tmpDir
```

Comment:
You don't need temp variable here, you can use directly: `gwCfg.MinioDir := ctx.Dir("minio")`